### PR TITLE
Revert "Remove top level functions for now"

### PR DESCRIPTION
**Pull request by brentleyjones**
This reverts commit b1cc2123c44d9763e5e7fb902181cff0dfa87fc8.

### CODE REVIEW

**Comment by brentleyjones** [comment]
Seems like buildifier doesn't handle the inline functions correctly.

File: swift/internal/module_maps.bzl [hunk 1]
```diff
@@ -91,18 +81,33 @@ def write_module_map(
...
+        return [
+            _header_path(
+                header_file = file,
+                relative_to_dir = relative_to_dir,
+                back_to_root_path = back_to_root_path,
+            )
+            for file in directory_expander.expand(file_or_dir)
+        ]
+
+    def _add_headers(*, headers, kind):
```
Comment (thii) [inline swift/internal/module_maps.bzl:94]: Not sure if you're already aware of but the latest of this function in the upstream branch has then been changed a bit (but we can update it in a follow-up PR).

https://github.com/bazelbuild/rules_swift/blob/c8b58b504842b6c16c2791e78db1137204f74a6d/swift/internal/module_maps.bzl#L103

**Review by thii — APPROVED** [review]

File: swift/internal/module_maps.bzl (hunk 1 above)
Comment (brentleyjones) [reply swift/internal/module_maps.bzl:94]: Was it changed in another commit that we haven't cherry-picked yet? I would rather grab it then.

Comment (thii) [reply swift/internal/module_maps.bzl:94]: https://github.com/bazelbuild/rules_swift/commit/868a76bed25439cf423871cc0803f58bdbf20d89 (already cherry-picked)

I manually fixed that with https://github.com/bazelbuild/rules_swift/commit/589e18fba0c21a702e1fe0e4b4bcc092c9672733

Comment (brentleyjones) [reply swift/internal/module_maps.bzl:94]: I'm matching https://github.com/bazelbuild/rules_swift/commit/868a76bed25439cf423871cc0803f58bdbf20d89 so far. Any other changes, like https://github.com/bazelbuild/rules_swift/commit/ef6662cec1fdb7731f2c11004a0831fc8b4faa33, I'll get when we cherry-pick those.

**Pull request merged**